apps/cowswap-frontend/src/modules/orders/utils/emitPostedOrderEvent.tsx

```diff
@@ -16,48 +16,34 @@ interface PendingOrderNotificationParams {
   inputAmount: CurrencyAmount<Token>
   outputAmount: CurrencyAmount<Token>
   orderCreationHash?: string
-  isHidden?: boolean
   isEthFlow?: boolean
 }
 
 export function emitPostedOrderEvent(params: PendingOrderNotificationParams) {
-  const {
+  const { chainId, id, receiver, owner, uiOrderType, orderCreationHash, inputAmount, outputAmount, isEthFlow } = params
+
+  const postedOrderPayload: OnPostedOrderPayload = {
+    orderUid: id,
+    orderCreationHash,
     chainId,
-    id,
-    receiver,
     owner,
-    uiOrderType,
-    orderCreationHash,
-    isHidden,
-    inputAmount,
-    outputAmount,
+    kind: params.kind,
+    orderType: uiOrderType,
+    inputAmount: BigInt(inputAmount.quotient.toString()),
+    outputAmount: BigInt(outputAmount.quotient.toString()),
+    inputToken: {
+      ...inputAmount.currency,
+      symbol: inputAmount.currency.symbol || '',
+      name: inputAmount.currency.name || '',
+    },
+    outputToken: {
+      ...outputAmount.currency,
+      symbol: outputAmount.currency.symbol || '',
+      name: outputAmount.currency.name || '',
+    },
+    receiver: receiver || undefined,
     isEthFlow,
-  } = params
-
-  if (!isHidden) {
```

Comment:
We actually don't use `isHidden` flag

Reply:
We do!
That's how we control whether a pre-sign order should be displayed or not.
By default, they start hidden and the notification is only emitted once the wallet replies.

Unless this has changed in your refactoring?

Reply:
But it doesn't work like that.

As far as I understand it should do:
When an order is created using smart-contract wallet, then we should not display the posted order popup.

In fact, we always display the popup. I've tested it with a version before the refactoring.